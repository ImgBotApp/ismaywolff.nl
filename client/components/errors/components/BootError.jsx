import React, { Component } from 'react'
import { string } from 'prop-types'
import ErrorContainer from './ErrorContainer'
import ErrorTitle from './ErrorTitle'
import { ExternalLink } from '../../links'
import { colors } from '../../../styles'

/**
 * Shown for errors that occur while booting / polyfilling.
 */

export class BootError extends Component {
  componentDidMount() {
    const error = new Error(`Boot error: ${this.props.errorMessage}`)

    import('../../../services/analytics')
      .then(analytics => analytics.trackError(error))
  }

  render() {
    return (
      <ErrorContainer background={colors.lightRed} color={colors.darkRed}>
        <ErrorTitle>Oops, something went wrong!</ErrorTitle>
        <p>This page uses modern javascript that your browser doesn&#39;t support and patching it
          has failed. Make sure that your browser extensions are not blocking the polyfill.io
          domain and refresh the page to retry or <ExternalLink href="http://browsehappy.com">update your browser</ExternalLink>.</p>
        <p>In most cases I&#39;ll be notified automatically that something went wrong and I&#39;ll
          make sure to fix it as soon as possible. If you want to report this error you can do so on
          the issues tab of my <ExternalLink href="https://github.com/ismay/ismaywolff.nl">github repo</ExternalLink>.
        </p>
        <p>The error was: &#34;{this.props.errorMessage}&#34;.</p>
      </ErrorContainer>
    )
  }
}

BootError.propTypes = {
  errorMessage: string.isRequired
}

export default BootError
