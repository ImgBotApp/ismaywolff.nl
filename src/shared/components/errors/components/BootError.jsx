import React from 'react'
import { string } from 'prop-types'
import { Link } from '../../link'
import { Title } from '../../text'
import ErrorWrapper from './ErrorWrapper'

/**
 * Shown for errors that occur while booting / polyfilling.
 */

const BootError = ({ errorMessage }) =>
  <ErrorWrapper type="error">
    <Title tag="h2" margin="0" center>
      Oops, something went wrong!
    </Title>
    <p>
      This page uses modern javascript that your browser doesn{"'"}t support and patching it has
      failed. Make sure that your browser extensions are not blocking the polyfill.io domain and
      refresh the page to retry or{' '}
      <Link href="http://outdatedbrowser.com">update your browser</Link>
      .
    </p>
    <p>
      In most cases I{"'"}ll be notified automatically that something went wrong and I{"'"}ll make
      sure to fix it as soon as possible. If you want to report this error you can do so on the
      issues tab of my <Link href="https://github.com/ismay/ismaywolff.nl">github repo</Link>
      .
    </p>
    <p>
      The error was: {'"'}
      {errorMessage}
      {'"'}.
    </p>
  </ErrorWrapper>

BootError.propTypes = {
  errorMessage: string.isRequired
}

export default BootError
