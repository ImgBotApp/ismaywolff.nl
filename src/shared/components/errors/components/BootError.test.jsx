/* global window, ga */

import React from 'react'
import { mount } from 'enzyme'
import BootError from './BootError'

window.ga = () => {}

describe('<BootError />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<BootError errorMessage={'Error message.'} />)
    expect(wrapper).toMatchSnapshot()
  })
})
