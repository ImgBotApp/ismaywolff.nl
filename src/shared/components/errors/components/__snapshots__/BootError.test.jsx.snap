// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BootError /> renders correctly 1`] = `
<styled.div
  type="error"
>
  <TitleFactory
    center={true}
    margin="0"
    size="1rem"
    tag="h2"
  >
    Oops, something went wrong!
  </TitleFactory>
  <p>
    This page uses modern javascript that your browser doesn
    '
    t support and patching it has failed. Make sure that your browser extensions are not blocking the polyfill.io domain and refresh the page to retry or
     
    <Link
      href="http://outdatedbrowser.com"
    >
      update your browser
    </Link>
    .
  </p>
  <p>
    In most cases I
    '
    ll be notified automatically that something went wrong and I
    '
    ll make sure to fix it as soon as possible. If you want to report this error you can do so on the issues tab of my 
    <Link
      href="https://github.com/ismay/ismaywolff.nl"
    >
      github repo
    </Link>
    .
  </p>
  <p>
    The error was: 
    "
    Error message.
    "
    .
  </p>
</styled.div>
`;
